### PR TITLE
Maven modules

### PR DESCRIPTION
**Pull request by jchambers**
Let me begin this pull request by pointing out that I have no practical experience with Maven modules, and any feedback here would be greatly appreciated. That said, the idea here is to break the project into modules so we can do things like add [microbenchmarks](http://openjdk.java.net/projects/code-tools/jmh/) and have a home for things like metrics libraries. Right now, there's just one module (the main project).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,76 +1,37 @@
...
     <description>A Java library for sending push notifications</description>
+
     <parent>
         <groupId>org.sonatype.oss</groupId>
         <artifactId>oss-parent</artifactId>
         <version>7</version>
     </parent>
+
+    <modules>
+        <module>pushy</module>
```
Comment (matzew) [inline pom.xml]: perhaps name it `pushy-core`, while others could be `pushy-metrics` etc ?

File: pushy/pom.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <artifactId>pushy</artifactId>
+    <name>Pushy</name>
+
+    <parent>
+        <groupId>com.relayrides</groupId>
+        <artifactId>pushy-parent</artifactId>
+        <version>0.7.2-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
```
Comment (matzew) [inline pushy/pom.xml]: It might be that some are being used on the other moduels, in that case I'd move them up, again .

Comment (jchambers) [inline pushy/pom.xml]: Agreed. I'm inclined to deal with that when it actually happens, though. Does that seem reasonable to you?

Comment (matzew) [inline pushy/pom.xml]: yes, why 'expose' all to all submodules :)

File: pom.xml (hunk 1 above)
Comment (jchambers) [inline pom.xml]: Fair enough. Done in 847954e.

**Comment by matzew** [comment]
👍

**Comment by matzew** [comment]
one thing. when the artifact is now named `-core` - perhaps version bump to 0.8.0 ?

**Comment by jchambers** [comment]
> when the artifact is now named `-core` - perhaps version bump to 0.8.0 ?

This could be me misunderstanding, but I don't think changing the module name necessarily changes the `artifactId`, right? As far as I can tell, the `pushy-core` *module* still produces an *artifact* called `pushy`.

File: pom.xml (hunk 1 above)
Comment (jchambers) [inline pom.xml]: Actually, I think I changed my mind here if only for the petty reason that having everything start with `pushy-` was making autocomplete slightly annoying ;)

Besides—we can always change it later.

**Comment by matzew** [comment]
sounds good

On Sun, May 8, 2016 at 5:51 PM, Jon Chambers <notifications@github.com>
wrote:

> Merged #284 <https://github.com/relayrides/pushy/pull/284>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/relayrides/pushy/pull/284#event-653738261>
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Pull request merged**